contracts/DomainRegistry.sol

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.20;
+
+contract DomainRegistry {
+    uint256 constant REQUIRED_DEPOSIT = 1 ether;
+
+    struct Domain {
+        uint256 deposit;
+        bool isRegistered;
+    }
+
+    mapping(string => Domain) public domains;
+
+    event DomainCreated(string tld, uint256 deposit);
+    event DomainReleased(string tld);
+
+    modifier hasRequiredDeposit(uint256 _requiredDeposit) {
+        require(msg.value >= _requiredDeposit, "Wrong eth amount");
+        _;
+    }
+
+    modifier domainDoesNotExist(string memory _tld) {
```

Comment:
It `_tld` the best arg name choice for someone who reads it for the first time?
-1 points

Reply:
fxd